### PR TITLE
Handle node_modules/.bin (#61)

### PR DESCRIPTION
**Pull request by rstacruz**
See #61.

### CODE REVIEW

File: test/integration/install.js [hunk 1]
```diff
@@ -39,8 +40,24 @@ describe('install', function () {
           var actualVersion = require(name + '/package.json').version
           assert(semver.satisfies(actualVersion, version), actualVersion + ' should satisfy' + version)
```
Comment (just-boris) [inline test/integration/install.js]: I'd like to put existence check here rather than doing an extra test.

File: lib/install.js [hunk 2]
```diff
@@ -120,6 +120,11 @@ function handleResolvedPkg (dir, pkg, cb) {
       // package level.
       fs.rename.bind(null, tmpDest, finalDest)
     ], function (err) {
+      // Include package.json's metadata.
+      // `pkg` at this point just comes from the resolver, which only has
+      // { name, version, uid, shasum, tarball }.
+      pkg = assign({}, require(path.join(dir, pkg.uid, 'package.json')), pkg)
```
Comment (just-boris) [inline lib/install.js]: I thought about another solution here.

We already have the `readPackage` function, here we can reuse its result.

Look at [this](https://github.com/alexanderGugel/ied/blob/master/lib/install.js#L48), here the `pkg` argument has proper information, then we need to pass it up to callback:

```js
function installSubDeps (tmpDest, dir, pkg, cb) {
  var subDependencies = assign({}, pkg.dependencies, pkg.peerDependencies)
  async.forEachOf(subDependencies, function (version, name, cb) {
    // ... cut ...
  }, function(err, result) {
      cb(err, result[0])
  })
}
```

Then we will have got package available [here](https://github.com/alexanderGugel/ied/blob/master/lib/install.js#L122) as the second argument.

**Comment by just-boris** [comment]
Thanks for you participation @rstacruz

File: lib/install.js (hunk 2 above)
Comment (rstacruz) [inline lib/install.js]: aye, that seems better. do a PR of that!

Comment (rstacruz) [inline lib/install.js]: btw, it seems my implementation only performs binstub linking on top-level dependencies... I _think_ this is consistent with npm behavior. doing it at this level would install binstubs for deep dependencies, wouldn't it?

Comment (just-boris) [inline lib/install.js]: No, my point is only about avoiding extra file reads. The behavior would be the same

Comment (rstacruz) [inline lib/install.js]: > Then we will have got package available here as the second argument.

The second argument there is:

```
[ [ '/..snip../ied/test/integration/node_modules/.tmp/a2caa045-3b99-42bc-ba35-0ccc1576bc10',
    [ null, undefined ],
    undefined ],
```

Comment (just-boris) [inline lib/install.js]: Sorry, maybe I missed something. 

That's why I was going to fix it at weekend, but not in between my main job duty :smile:

**Comment by just-boris** [comment]
I think this PR is superseded by https://github.com/alexanderGugel/ied/issues/61
@rstacruz please check, did my fix worked for you?

**Comment by rstacruz** [comment]
too lazy to check but I trust you man haha.

**Pull request closed without merging**